src/axom/sidre/core/Array.hpp

```diff
@@ -60,7 +60,7 @@ constexpr axom::IndexType ZERO = 0;
  * \see View
  */
 template< typename T >
-class Array : public axom::Array<T>
+class Array : public axom::mint::Array<T>
```

Comment:
I believe we discussed moving the old Array to `axom::core::MCArray` or something like that as a first step. I don't think we want Sidre to depend on Mint b/c that creates a circular dependency. Would that be feasible in this PR ?

Reply:
@gzagaris -- this is a good catch. We absolutely do not want `sidre` to depend on `mint`.

We had originally discussed a two-pronged approach:
* First: Moving the current `axom::Array` int `mint` to minimize the disruption for this initial change
* Later: Creating some form of multi-component array (`MCArray`)  for sidre and mint that is implemented in terms of Brian's new `axom::Array` class

I don't think we considered that the current `sidre::Array` depends on the current `axom::Array`.
Since it does, it sounds like the best short-term approach would be to follow your suggestion.
I.e. 
* move `mint::Array` back to `core` and rename it `MCArray`
* update `sidre:Array` to derive from `MCArray`

@bmhan12 -- Does that sound reasonable?

Reply:
Yes, I think this makes sense, and I'll have this change put in this PR.